JSDemos/Demos/PivotGrid/SimpleArray/description.md

```diff
@@ -1 +1,8 @@
-The PivotGrid is&nbsp;a&nbsp;high-performance, enterprise-ready UI component optimized for multi-dimensional data analysis. It&nbsp;allows your end users to&nbsp;easily arrange and summarize information in&nbsp;a&nbsp;cross-tabular format. In&nbsp;this demo, the PivotGrid&rsquo;s simplicity of&nbsp;use is&nbsp;illustrated. Sorting and filtering are enabled so&nbsp;you can sort data in&nbsp;ascending or&nbsp;descending order and by&nbsp;summary values. You can drag and drop fields to&nbsp;reorganize the PivotGrid and change filtering properties to&nbsp;analyze data more efficiently.
\ No newline at end of file
+The PivotGrid component can display data from an array of objects. Use the [dataSource](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/#dataSource) property to bind the PivotGrid to data. This property accepts a [PivotGridDataSource](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/) instance or its configuration object. Assign your array to this object's [store](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/store/) property.
```

Comment:
> The PivotGrid component can display data **from an array of objects**.

What does it mean? Can I use any objects as data sources? Can I create custom data sources? Are there any requirements for this? Consequent sentences clarify this for a certain degree but not quite.

> Use the [dataSource](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/#dataSource) property to bind the PivotGrid to data.

OK. No questions here. However...

> This property accepts a [PivotGridDataSource](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/) instance or **its configuration object**.

What type the mentioned "configuration object" has? Does it stem from the same root (base class) as `PivotGridDataSource`? What if I need both the data source and its configuration object? Should I assign the data source to the mentioned property first and the configuration object second?

> Assign your array to this object's [store](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/store/) property.

OK. This sentence clearly has close ties with the first sentence in this paragraph. However, this does not make it easier to understand the connection between "an array of objects" and `PivotGridDataSource`. Do they complement each other or do they serve as alternative options? If the former is correct, then how do they interact with each other?

Reply:
Hello. Seems like technologies you work every day with have much more strict typing than JavaScript. This causes most of the questions. Let's start from the very beginning:

>What does it mean? Can I use any objects as data sources? Can I create custom data sources? Are there any requirements for this? Consequent sentences clarify this for a certain degree but not quite.

This demo is completely dedicated to a simple array of objects. Yes, you can have any number of fields of any types in these objects since you specify the required fields in the fields[] array as we say below. Not sure that there are any strict requirements or limitations. You just pass data and take the fields you need.

>What type the mentioned "configuration object" has? 

Sorry, let's start from the basics. Most components require that you assign a [DataSource ](https://js.devexpress.com/Documentation/ApiReference/Data_Layer/DataSource/) instance to the [dataSource ](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#dataSource) property. DataSource is the object itself, but you can get it from the function or a variable. This is the reason why we distinguish DataSource and its configuration object.

>Does it stem from the same root (base class) as PivotGridDataSource? What if I need both the data source and its configuration object?

As I said before, DataSource (and PivotGridDataSource) is an object. You can check its possible structure here: https://js.devexpress.com/Documentation/ApiReference/Data_Layer/PivotGridDataSource/. 

>Should I assign the data source to the mentioned property first and the configuration object second?

I hope that I answered this question in the previous paragraphs. If not, please pm me in teams.

>OK. This sentence clearly has close ties with the first sentence in this paragraph. However, this does not make it easier to understand the connection between "an array of objects" and PivotGridDataSource. Do they complement each other or do they serve as alternative options? If the former is correct, then how do they interact with each other?

The data source's structure is the following:
you have an array of objects (let's take the one from the data.js [in this demo](https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/SimpleArray/jQuery/Light/) that is called 'sales`). 
![image](https://user-images.githubusercontent.com/33040934/125633714-11fa6665-ca6f-4a0e-8454-bba17c72a0b9.png)
You should assign it to the `store` field of the PivotGrid's dataSource object:
![image](https://user-images.githubusercontent.com/33040934/125633816-42832f52-9191-4d51-b92b-cc391147bbec.png)
Now you have the data and can display it. To modify it in some ways (set the required fields, filter, delete...), use the configuration properties described in the [PivotGridDataSource ](https://js.devexpress.com/Documentation/ApiReference/Data_Layer/PivotGridDataSource/) section.

Reply:
> Hello. Seems like technologies you work every day with have much more strict typing than JavaScript.

Hello. Yes, you are absolutley right. JavaScript indeed seems like a completely alien world to me. Therefore, you've recieved a review from a perfect stanger, be that good or bad. However, if we speak of an array of objects, would an array of integers comply to this definition, for example? Normally, I would expect that if I assign something else to a property rather that that would be expected, I'll get an error, an exception, or undefined behavior, at the very least. That's why my question about limitations and requirements is natural, from my point of view.

> Most components require that you assign a DataSource instance to the dataSource property. DataSource is the object itself, but you can get it from the function or a variable.

Yes, I get this, A function or variable can return a data source instance that you can assign to a property but... What does it have to do with a configuration object? How should I define it? To which property should I assign it? Again, I apologize if this is something very obvious to your customers. I only see only something weird and unintelligible.

Can I use a data source's property to access the configuration object? Is it a sort of the data source's property set?

Reply:
A summary from our personal discussion:

The mentioned pivot grid data source and configuration object are two different ways to define a data source. The data source instance is an abstract object that can work with data in any form, including an array of abstract objects.

The goal here is to create a holistic perception of the data source functionality in text. In its current shape, the text describes the mentioned data array, pivot grid data source, and configuration objects as things that have absolutely nothing in common. For instance, you need to mention that the `store` property belongs to the pivot grid data source.